docs/en/guides/flakes-guide.md

```diff
@@ -0,0 +1,128 @@
+
+## How can I use flakes?
+
+To begin with using flakes, you need to enable flakes on your system. This can be done by editing `/etc/nix/nix.conf` or your NixOS configuration.
+
+For non-NixOS systems, add the following to `/etc/nix/nix.conf` or `~/.config/nix/nix.conf`:
+
+```
+experimental-features = nix-command flakes
+```
+
+For NixOS systems, add the following to your NixOS configuration:
+
+```nix
+# configuration.nix
+nix.settings.experimental-features = [ "nix-command" "flakes" ];
+```
+
+Next, initialise a Nix flake using the `nix flake init` command. This will create a basic flake.nix and flake.lock, which should look like this:
+
+```nix
+# flake.nix
+{
+  description = "A very basic flake";
+
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs?ref=nixos-unstable";
+  };
+
+  outputs = { self, nixpkgs }: {
+
+    packages.x86_64-linux.hello = nixpkgs.legacyPackages.x86_64-linux.hello;
+
+    packages.x86_64-linux.default = self.packages.x86_64-linux.hello;
+
+  };
+}
+```
+
+The default flake template includes a description, a list of inputs including nixpkgs, and a set of packages as outputs. There are many different possible types of outputs, which are all used by different nix commands and projects. In this instance, the `packages.x86_64-linux.default` output is used by the `nix build .` command, which builds the package and adds it to the nix store. The most common output type, and the most important one for this guide, is `nixosConfiguration."<hostname>"`. This outputs a NixOS configuration which is used by `nixos-rebuild switch`. Here is an example of a basic flake for a NixOS configuration:
+
+```nix
+# flake.nix
+{
+  description = "A very basic flake";
+
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs?ref=nixos-unstable";
+  };
+
+  outputs = { self, nixpkgs, ... } @ inputs: {
+    nixosConfiguration.nixos-system = {
+      specialArgs = {inherit inputs;};
+      modules = [
+        ./configuration.nix
+      ];
+    };
+  };
+}
+```
+
+This file includes a few new parts. First, the `{ self, nixpkgs }` has been changed to `{ self, nixpkgs, ... } @ inputs`. This allows for the flake to recognise any inputs passed in, rather than only being able to use nixpkgs. Next, the packages have been replaced with a `nixosConfiguration.nixos-system`. This exposes a new nixos configuration with the hostname nixos-system to `nixos-rebuild`. When using this in a real system, make sure to change nixos-system to whatever your system hostname is. Thirdly, the `specialArgs = {inherit inputs;};` allows other files to access the flake inputs. Finally, the modules list includes every file imported into the NixOS config. This file can be placed alongside an existing NixOS configuration to add flakes to it. Using `nixos-rebuild switch` will create a flake.lock file and begin using flakes.
```

Comment:
> This file includes a few new parts. First, the `{ self, nixpkgs }` has been changed to `{ self, nixpkgs, ... } @ inputs`. This allows for the flake to recognise any inputs passed in, rather than only being able to use nixpkgs.

The main "benefit" of `@ inputs` is to destructure the argument set. In addition, you may very well add more inputs to the argset (e.g. home-manager) without ever destructuring it.

The first sentence is incorrect.

> When using this in a real system, make sure to change nixos-system to whatever your system hostname is

This does not have to be the hostname. The name is just to identify the contents of the set, nothing more. A system's hostname must be set with `networking.hostName`.